cloudify_ansible/tasks.py

```diff
@@ -24,20 +24,31 @@
 from cloudify_ansible import (
     constants,
     ansible_playbook_node,
-    ansible_relationship_source
+    ansible_relationship_source,
+    utils
 )
 
 
 @operation
 @ansible_playbook_node
-def run(playbook_args, _ctx, **_):
+def run(playbook_args, ansible_env_vars, _ctx, **_):
     _ctx.logger.info('playbook_args: {0}'.format(playbook_args))
     try:
-        result = AnsiblePlaybookFromFile(**playbook_args).execute()
-        _ctx.logger.debug('result: {0}'.format(result.__dict__))
-        _ctx.instance.runtime_properties['result'] = result.__dict__
+        playbook = AnsiblePlaybookFromFile(**playbook_args)
     except CloudifyAnsibleSDKError:
         raise NonRecoverableError(CloudifyAnsibleSDKError)
+    # Because Ansible is written as a command-line tool,
+    # there are some options that only perform well as
+    # environment variables.
+    utils.assign_environ(ansible_env_vars)
+    result = playbook.execute()
+    utils.unassign_environ(ansible_env_vars)
+    utils.handle_result(
```

Comment:
@EarthmanT Is there a case that result will return `None` at this case ?

Reply:
should not ever happn.